tests/utils/waitForPromisesToResolve.js

```diff
@@ -0,0 +1,8 @@
+/**
+ * Method which waits for all asynchronous JS to stop executing before proceeding. This helps test things like actions
+ * that expect some Onyx value to be available. This way we do not have to explicitly wait for an action to finish
+ * (e.g. by making it a promise and waiting for it to resolve).
+ *
+ * @returns {Promise}
+ */
+export default () => new Promise(setImmediate);
```

Comment:
TIL about `setImmediate`. It looks to be pretty non-standard method though from looking at the MDN docs. Maybe it would be better to use `setTimeout()` like this?

```
new Promise((resolve) => {
    setTimeout(resolve, 0);
});
```

Reply:
Good thought. I think maybe that will work too? Then again I also feel OK about keeping this since `setImmediate` is part of `Node` and that's the environment jest runs in.

Reply:
I tested and the `setTimeout()` does not work 😢

Reply:
Huh... weird. Yeah, I think it's fine if this is always running in Node. I was thinking that maybe at some point someone decides to switch these tests to QUnit which would run in the browser and might cause some issues where it could be hard to track down. That sounds like over-engineering it though, so I'm OK leaving it like this.